src/main/java/org/mockito/Mockito.java

```diff
@@ -1278,6 +1278,50 @@
      * </code></pre>
      */
     public static final Answer<Object> CALLS_REAL_METHODS = Answers.CALLS_REAL_METHODS;
+
+    /**
+     * Optional <code>Answer</code> to be used with {@link Mockito#mock(Class, Answer)}.
+     *
+     * In the example below a Builder is mocked. Normally, when {@code builder.doSomething()} is called, it returns this.
+     * In order to prevent the builder of returning null (default answer), it will return itself.
+     *
+     * <pre class="code"><code class="java">
+     * Builder builder = mock(Builder.class, RETURNS_SELF);
+     *
+     * assertEquals(builder, builder.doSomething());
+     * </code></pre>
+     *
+     * <p><b>Keep in mind this answer uses the return type of a method.
+     * If this type is assignable to the class of the mock, it will return the mock.
+     * Therefore if you have a method returning a superclass (for example {@code Object}) it will match and return the mock.</b></p>
+     *
+     * The builder created in ExtendedBuilderTest will <b>on every method call</b> return itself.
+     * <pre class="code"><code class="java">
+     * public class Builder {
+     *
+     *     public Builder append() { ... }
+     *
+     *     public Object appendThis() { ... }
+     * }
+     *
+     * public class ExtendedBuilder extends Builder {
+     *
+     *     public Builder extend() { ... }
+     *
+     *     public ExtendedBuilder extendThis() { ... }
+     * }
+     *
+     * public class ExtendedBuilderTest {
+     *
+     *     &#064;Test
+     *     public void mockBuilder() {
+     *         ExtendedBuilder builder = mock(ExtendedBuilder.class, RETURN_SELF);
+     *     }
```

Comment:
Also the example should show how to use (stub/verify) the terminal method of a builder (`.make()`, etc.)